### ISSUE TITLE
[BUG] pdfjs-dist loader issue [Update version]

### ISSUE THREAD

**Issue by prafulrajhm** [ansu5555/pdf-viewer-reactjs#87]
When I update the package to the latest version, I am getting the below error

![Code_7bkZS7zwvW](https://user-images.githubusercontent.com/44769213/142853506-e650427d-7d40-45f5-b517-f7b061a80204.png)

**How to Reproduce**
1. Check next js app with all latest versions (Mentioned below)

**Expected behavior**
Should work!

**Details (please complete the following information):**
 - Device:  Desktop
 - OS:  Windows 10
 - Browser: chrome(96.0.4664.45)
 - Version: 2.2.3 , React: 16.14.0

**Additional context**
https://github.com/mozilla/pdf.js/issues/13200

**Comment by Luiscastor** [comment]
hello, i'm having the same problem. did you find a solution for this?

**Comment by prafulrajhm** [comment]
> hello, i'm having the same problem. did you find a solution for this?

No, I used 2.1.2 only which requires 2.5.207 version of pdfjs-dist.

**Comment by Luiscastor** [comment]
Okey error is gone but now i got this one Error while reading the pages ! :(

**Comment by prafulrajhm** [comment]
> Okey error is gone but now i got this one Error while reading the pages ! :(

It can be of number of reasons! check the console. 

May be pdf file is wrong or the way you are passing the props.

**Comment by Luiscastor** [comment]
All good, thank u so much bro! I'm gonna be working with this problem jeje xd

**Comment by mepritam** [comment]
I had the same issue. Just downgrade **pdfjs-dist to 2.6.347**. 
Run `npm install pdfjs-dist@2.6.347`

**Comment by Luiscastor** [comment]
Hello sr, I do not know what it's happening. I downgrade pdfjs-dist to 2.6.347. but I'm getting the first error,
the one that you post.

**Comment by mepritam** [comment]
> Okey error is gone but now i got this one Error while reading the pages ! :(

Which one ?

**Comment by Luiscastor** [comment]
<img width="860" alt="Captura de Pantalla 2021-12-03 a la(s) 21 45 40" src="https://user-images.githubusercontent.com/36940666/144695767-b6e60b90-4692-4ddb-828c-c60c5ec325bb.png">
this one bro.

**Comment by swingtechnologies** [comment]
The below solution works for me
First, install
npm install npm-force-resolutions --save-dev

Then in package.json we need to override the pdfjs-dist version to previous working version like this

```
"dependencies": {
   "pdf-viewer-reactjs": "2.2.3",
}.
"scripts": {
   "preinstall": "npx npm-force-resolutions"
},
"resolutions": {
    "pdfjs-dist": "2.6.347"
 }
```
and in scripts section add new preinstall entry:

"preinstall": "npx npm-force-resolutions",
now if you do npm install and run your application it should work. Hope it helps

The root cause of the problem is , though in the dependency package.json has version 2.6.347 for pdfjs-dist since ^ was there before the version number during npm install, it installs the most recent version of pdfjs-dist which is causing this problem.

**Comment by WalissonRodrigo** [comment]
I'm found this problem and solve using **useMemo** to generate object url. This [commet](https://github.com/ansu5555/pdf-viewer-reactjs/issues/51#issuecomment-1063458281) has details and an example
